cmd/package_managers.go

```diff
@@ -138,3 +206,134 @@ func fetchGitRepositoryFromRubyGems(packageName string, manager packageManagerCl
 	}
 	return v.SourceCodeURI, nil
 }
+
+// Gets the GitHub repository URL for the nuget package.
+func fetchGitRepositoryFromNuget(packageName string, manager packageManagerClient) (string, error) {
+	nugetIndexURL := "https://api.nuget.org/v3/index.json"
+	respIndex, err := manager.GetURI(nugetIndexURL)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("failed to get nuget index json: %v", err))
+	}
+	defer respIndex.Body.Close()
+	nugetIndexResults := &nugetIndexResults{}
+	err = json.NewDecoder(respIndex.Body).Decode(nugetIndexResults)
+
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to parse nuget index json: %v", err))
+	}
+	nugetPackageBaseURL, err := getFieldFromIndexResults(nugetIndexResults.Resources,
+		"PackageBaseAddress/3.0.0")
+	if err != nil {
+		return "", err
+	}
+	nugetRegistrationBaseURL, err := getFieldFromIndexResults(nugetIndexResults.Resources,
+		"RegistrationsBaseUrl/3.6.0")
+	if err != nil {
+		return "", err
+	}
+
+	lowerCasePackageName := strings.ToLower(packageName)
+	lastPackageVersion, err := getLatestListedVersion(nugetRegistrationBaseURL,
+		lowerCasePackageName, manager)
+	if err != nil {
+		return "", err
+	}
+
+	respPackageSpec, err := manager.Get(
+		nugetPackageBaseURL+"%[1]v/"+lastPackageVersion+"/%[1]v.nuspec", lowerCasePackageName)
+	if err != nil {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to get nuget package spec json: %v", err))
+	}
+	defer respPackageSpec.Body.Close()
+
+	packageSpecResults := &nugetNuspec{}
+	err = xml.NewDecoder(respPackageSpec.Body).Decode(packageSpecResults)
+
+	if err != nil || packageSpecResults.Metadata == (nuspecMetadata{}) {
+		return "", sce.WithMessage(sce.ErrScorecardInternal,
+			fmt.Sprintf("failed to parse nuget nuspec xml: %v", err))
+	}
+
+	if packageSpecResults.Metadata.Repository != (nuspecRepository{}) &&
+		len(strings.TrimSpace(packageSpecResults.Metadata.Repository.URL)) != 0 {
+		return packageSpecResults.Metadata.Repository.URL, nil
+	}
+	if len(strings.TrimSpace(packageSpecResults.Metadata.ProjectURL)) != 0 &&
+		isSupportedProjectURL(packageSpecResults.Metadata.ProjectURL) {
+		return packageSpecResults.Metadata.ProjectURL, nil
+	}
+	return "", sce.WithMessage(sce.ErrScorecardInternal,
+		fmt.Sprintf("source repo is not defined for nuget package %v", packageName))
+}
+
+func isSupportedProjectURL(projectURL string) bool {
+	pattern := `^(?:https?://)?(?:www\.)?(?:github|gitlab)\.com/([A-Za-z0-9_\.-]+)/([A-Za-z0-9_\.-]+)$`
```

Comment:
nit: trailing slash should probably be acceptable. This URL appears to work fine:
https://github.com/ossf/scorecard/

Example: https://www.nuget.org/packages/awssdk.core has `https://github.com/aws/aws-sdk-net/`

Reply:
fixed and tested